[wz2k] iP

wz2k opened the pull request:
# TaskBotDuke

## Available functionalities
+ **create** tasks
+ **mark/umark** tasks
+ **delete** tasks
+ **find** tasks

## The following types of tasks are available:
1. Todos
2. Deadlines
3. Events

```java
todo read book
deadline return book /by 2023-05-20
event book sale /from 2023-01-30 /to 2023-03-06
```

> All tasks can be displayed with the list command

### Why should you be using TaskBotDuke? :thinking:
- [x] *simple and easy* to use
- [x] ~~bot~~ human interaction
- [x] low disk space requirement
- [x] dates are handled by Java's ```java.time.LocalDate```

Releases are available [here](https://github.com/wz2k/ip/releases)

Commit:
Add Gradle support

wz2k committed:
feat: chatbot greeting and echoing

wz2k committed:
docs: added javadocs

wz2k committed:
feat: storing text and displaying text

wz2k committed:
feat: marking tasks as done

wz2k committed:
feat: tracking for todos, deadlines and events

wz2k committed:
test: input and output

wz2k committed:
fix: extra whitespaces causing incorrect output

wz2k committed:
feat: responses to invalid inputs

wz2k committed:
docs: small wording changes

wz2k committed:
feat: delete tasks

wz2k committed:
Add saving of data locally

wz2k committed:
Add date handling for deadlines and events

wz2k committed:
Merge branch 'branch-Level-7'

wz2k committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

wz2k committed:
Refactor code into more classes

wz2k committed:
Refactor classes into packages

wz2k committed:
Merge remote-tracking branch 'origin/add-gradle-support'

wz2k committed:
Add spacing to deadline and event sentences

wz2k committed:
Add Junit tests for all task types

wz2k committed:
Update main class name

wz2k committed:
Update storage initialization

wz2k committed:
Add JavaDoc comments

wz2k committed:
Refactor code to meet coding standard

wz2k committed:
Add searching for tasks using keywords

wz2k committed:
Merge branch 'branch-A-JavaDoc'

wz2k committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/Todo.java

wz2k committed:
Merge branch 'branch-Level-9'

wz2k committed:
Update JavaDocs

wz2k committed:
Add checkstyle to functionality to gradle

wz2k committed:
Clean up checkstyle errors

wz2k committed:
Update README.md

wz2k committed:
Update README.md to include syntax highlighting

wz2k committed:
Update README.md to group up code

wz2k committed:
Add a GUI for duke

wz2k committed:
Merge branch 'branch-Level-10'

wz2k committed:
Merge branch 'master' of https://github.com/wz2k/ip

wz2k committed:
Add index numbering when listing tasks

PawanPatil19 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,77 @@
+     *
+     * @return Task marked symbol.
+     */
+    private String isMarkedSymbol() {
```
As it not returning a boolean, perhaps it could have a 'get' verb in in the method name

PawanPatil19 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,95 @@
+        int size = taskList.size();
+
+        if (size == 0 || taskNumber > size) {
+            throw new DukeException("duke.task.Task number does not exist.");
```
Any reason the error message is this way?

wz2k replied:
Thanks for spotting this, might be due to intellij's auto refactoring when I was renaming

PawanPatil19 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,45 @@
+     */
+    public String getResponse(String input) {
+        try {
+            Command command = Parser.parseCommand(input, taskList, storage);
+            return command.execute();
```
I like the way you have seperated parsing and executing the commands 👍

PawanPatil19 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,27 @@
+ *
+ * @author wz2k
+ */
+public abstract class Command {
```
I like the way you have implemented this abstract class 👍 .

PawanPatil19 reviewed (commented):
LGTM..

tangyuantyy commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
```
I like that your header comments are clear and concise!

tangyuantyy commented on src/main/java/duke/command/ListTasksCommand.java:
```diff
@@ -0,0 +1,49 @@
+     * Lists the tasks for the user and returns the reply with the list
+     * of found tasks
+     *
+     * @return the list of found tasks.
```
```suggestion
     * @return The list of found tasks.
```

wz2k replied:
Thanks for helping to spot that small error 👍

tangyuantyy reviewed (commented):
LGTM! Overall, I found your code really clean and easy to read

wz2k committed:
Add assertions to enforce program state

Command messages should have the correct amount of
dividers so assertions are used to check if there
is a correct amount of parts after splitting the
command messages.

wz2k committed:
Refactor code to improve code quality.

Added default case to switch statements, renamed
a few variables and practiced some SLAP to make
code more readable

wz2k committed:
Merge pull request #1 from wz2k/branch-A-Assertions

Add assertions to enforce program state

wz2k committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/command/CreateDeadlineCommand.java
#	src/main/java/duke/command/CreateEventCommand.java
#	src/main/java/duke/command/CreateTodoCommand.java
#	src/main/java/duke/command/DeleteTaskCommand.java
#	src/main/java/duke/command/MarkTaskCommand.java
#	src/main/java/duke/command/UnmarkTaskCommand.java

wz2k committed:
Merge pull request #2 from wz2k/branch-A-CodeQuality

Refactor code to improve code quality.

wz2k committed:
Refactor code to improve code quality.

Added line breaks after assert to separate code
logic.

wz2k committed:
Set up Continuous Integration for IP

Added a yml file

wz2k committed:
Reconfigure file to yml file

wz2k committed:
Add reminders by chatbot on launch

wz2k committed:
Merge branch 'branch-BCD-Extension'

wz2k committed:
Clean up code

wz2k committed:
Improve GUI

wz2k committed:
Renamed folder

wz2k committed:
Clean up build gradle file

wz2k committed:
Regenerate gradle wrapper

wz2k committed:
Add user guide

wz2k committed:
Fix spelling errors

wz2k committed:
Add invalid input handling

wz2k committed:
Change javaFX api version to remove warning

wz2k committed:
Clean up code

wz2k committed:
Fix user guide

wz2k committed:
Fix user guide spacing issues

wz2k committed:
Fix user guide code blocks
